packages/framework/fluid-static/src/types.ts

```diff
@@ -58,9 +58,9 @@ export type DataObjectClass<T extends IFluidLoadable> = {
  * @typeParam T - The class of the `SharedObject`.
  * @public
  */
-export type SharedObjectClass<T extends IFluidLoadable> = {
-	readonly getFactory: () => IChannelFactory;
-} & LoadableObjectCtor<T>;
+export interface SharedObjectClass<T extends IFluidLoadable> {
+	readonly getFactory: () => IChannelFactory<T>;
```

Comment:
Docs?

Reply:
This whole API surface seems a bit screwy. Here we have an interface thats implemented in packages that arn't allowed to depend on this interface.

Also the interface is a factory factory for what seems like no reason (could be just the IChannelFactory).

This could really use another follow-up breaking change that rationalizes and document is. I'll minimally patch some of the docs gaps for now in this change.

Reply:
I've added some docs, though documenting his is hard since it doesn't make a lot of sense to me.